kore/src/Kore/Step/AntiLeft.hs

```diff
@@ -0,0 +1,417 @@
+                , maybeInner = forgetSimplified <$> maybeInner
+                , leftHands = map forgetSimplifiedLeft leftHands
+                }
+
+forgetSimplifiedLeft
+    :: InternalVariable variable
+    => AntiLeftLhs variable
+    -> AntiLeftLhs variable
+forgetSimplifiedLeft antiLeftLhs@(AntiLeftLhs _ _ _ _) = case antiLeftLhs of
+    AntiLeftLhs {aliasTerm, existentials, predicate, term} ->
+        AntiLeftLhs
+            { aliasTerm = TermLike.forgetSimplified aliasTerm
+            , existentials
+            , predicate = Predicate.forgetSimplified predicate
+            , term = TermLike.forgetSimplified term
+            }
+
+toTermLike :: AntiLeft variable -> TermLike variable
+toTermLike AntiLeft {aliasTerm} = aliasTerm
+
+{-
+Supported syntax:
+antiLeft = antiLeftAlias
+
+The antiLeftAlias expands to
+or(nextAntiLeft, or(lhs1, or(lhs2, ... or(lhsn, bottom) ... )))
+where nextAntiLeft is optional.
+
+nextAntileft has the same syntax as antiLeft.
+lhs is of the form
+exists x1 . exists x2 . ... exists xn . lhsTermAlias(x1, ..., xn)
+
+lhsTermAlias expands to
+and(lhsPredicate, lhsTerm)
+-}
+parse
+    :: InternalVariable variable
+    => TermLike variable -> Maybe (AntiLeft variable)
+parse aliasTerm@(ApplyAlias_ alias params) = do
+    (maybeInner, lhss) <-
+        case substituteInAlias alias params of
+            substituted@(Or_ _ first remaining) ->
+                case parse first of
+                    Just nextAntiLeft ->
+                        Just (Just nextAntiLeft, remaining)
+                    Nothing -> Just (Nothing, substituted)
+            _ -> Nothing
+    leftHands <- parseLhss lhss
+    return AntiLeft {aliasTerm, maybeInner, leftHands}
+parse _ = Nothing
+
+parseLhss
+    :: InternalVariable variable
+    => TermLike variable -> Maybe [AntiLeftLhs variable]
+parseLhss (Or_ _ first nexts) = do
+    firstParsed <- parseLhs first
+    nextsParsed <- parseLhss nexts
+    return (firstParsed : nextsParsed)
+parseLhss (Bottom_ _) = Just []
+parseLhss _ = Nothing
+
+parseLhs
+    :: InternalVariable variable
+    => TermLike variable
+    -> Maybe (AntiLeftLhs variable)
+parseLhs lhs = case aliasTerm of
+    (ApplyAlias_ alias params) -> case substituteInAlias alias params of
+        (And_ _ predicate term) ->
+            Just AntiLeftLhs
+                { aliasTerm
+                , existentials
+                , predicate = Predicate.wrapPredicate predicate
+                , term
+                }
+        _ -> Nothing
+    _ -> Nothing
+  where
+    (existentials, aliasTerm) = stripExistentials lhs
+
+    stripExistentials
+        :: TermLike variable -> ([ElementVariable variable], TermLike variable)
+    stripExistentials (Exists_ _ var term) = (var : vars, remaining)
+      where
+        (vars, remaining) = stripExistentials term
+    stripExistentials term = ([], term)
+
+{-| Creates the AntiLeft predicate as described in
+docs/2020-06-30-Combining-Priority-Axioms.md
+-}
```

Comment:
In the document, I think what you have below appears negated, is that right?

Reply:
I added some explanations, hopefully it's more clear now.